test/application/use.js

```diff
@@ -52,4 +53,11 @@ describe('app.use(fn)', function(){
     .expect(404)
     .end(done);
   });
+
+  it('should throw error for non function', function(done){
+    const app = new Koa();
+
+    should(function(){ app.use('not a function'); }).throw('middleware must be a function!');
```

Comment:
Also, I have some free time this morning, do you need test function refactor using arrow function like:
it('should', done => {...}) ?